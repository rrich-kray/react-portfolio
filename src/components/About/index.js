import React from "react";

const About = () => {
  return (
    <div className="about-container d-flex flex-column w-100 justify-content-center align-items-center">
      <div className="left-panel panel justify-content-center align-items-center d-flex">
        <div className="pfp-container">
          <img src={require("./avatar2.jpg")} className="pfp" alt="pfp"></img>
        </div>
      </div>
      <div className="right-panel panel d-flex justify-content-center align-items-center">
        <div className="bio-container">
          <h1 data-testid="h1test" className="about-title">
            About Me
          </h1>
          <p data-testid="paragraph-test" className="about-text-container">
            My name is Ryan, and I am currently a student attending the Case
            Western Reserve Full Stack Web Development Bootcamp. My career began
            in equity research, where I produced research for institional
            investor clients such as pension funds. Here, I frequently used
            Python to build and organize large proprietary datasets that often
            served as the bedrock for differentiated, revenue-generating
            research. More importantly, it helped me realize my passion for
            software development and motivated my departure from the industry
            nearly a year ago. Since then, I have built dozens of projects
            leveraging technologies across the entire MERN stack, and I continue
            to be motivated by the prospect of developing unique solutions to
            unique problems.
          </p>
        </div>
      </div>
    </div>
  );
};

export default About;
