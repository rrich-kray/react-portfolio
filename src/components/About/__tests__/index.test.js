import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import About from "../index";

afterEach(cleanup);

describe("Component renders", () => {
  // renders component
  it("renders", () => {
    render(<About />);
  });

  it("Matches the DOM node structure", () => {
    const { asFragment } = render(<About />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Renders About Me element", () => {
    const { getByTestId } = render(<About />);
    expect(getByTestId("h1test")).toHaveTextContent("About Me");
  });
});
