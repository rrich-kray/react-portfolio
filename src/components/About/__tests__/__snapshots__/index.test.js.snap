// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders Matches the DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="about-container d-flex w-100 justify-content-center align-items-center"
  >
    <div
      class="left-panel justify-content-center align-items-center d-flex"
    >
      <div
        class="pfp-container"
      >
        <img
          alt="pfp"
          class="pfp"
          src="avatar.svg"
        />
      </div>
    </div>
    <div
      class="right-panel d-flex justify-content-center align-items-center"
    >
      <div
        class="bio-container"
      >
        <h1
          class="about-text-container"
          data-testid="h1test"
        >
          About Me
        </h1>
        <p
          data-testid="paragraph-test"
        >
          Veniam ex reprehenderit consequat ex quis ad quis aliquip. Voluptate sunt esse et eiusmod ut ullamco amet. Laboris anim ullamco cupidatat laborum do consequat tempor sint occaecat ad. Pariatur ut qui esse minim quis dolore enim id tempor deserunt. Sint ullamco occaecat cillum magna nisi. Ipsum quis irure enim incididunt proident dolor ad ad enim esse incididunt esse. Lorem velit laboris nostrud dolor labore eu irure reprehenderit labore. Reprehenderit ad velit labore eu veniam voluptate amet tempor est Lorem aliquip ea excepteur. Ipsum irure velit excepteur duis elit dolor occaecat culpa dolore excepteur. Aliqua excepteur sunt voluptate veniam consectetur. Magna enim ad et sit do sit dolore esse ea. Eu ut excepteur tempor eiusmod in. Exercitation elit do esse dolor. Velit adipisicing est ex ad esse deserunt officia proident adipisicing magna incididunt. Incididunt nisi non veniam deserunt qui laboris eu quis.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
